### ISSUE TITLE
Ensure GVFS provider only accesses .git on lower storage filesystem

### ISSUE THREAD

**Issue by chrisd8088** [github/VFSForGit#15]
At present, mounting the GVFS provider stalls for a potentially long period of time because the initial enumeration callback for the top-level directory of the mount point (`src/`) iterates through the list of entries in that directory, and for each one, tries to read the "blob state" by [looking](https://github.com/Microsoft/VFSForGit/blob/a0d7234095713d2a64165fced5dc369715bbbdeb/GVFS/GVFS.Common/Git/GitRepo.cs#L225) in `src/.git` (which is the path currently returned by `this.enlistment.LocalObjectsRoot`).  Each of these requests goes to libprojfs, which tries to enumerate the parent directory, which (as intended) times out waiting for the per-inode [hydration lock](https://github.com/github/libprojfs/blob/f0b39cc7a725138290097fb469c169e23081a879/lib/projfs.c#L248).  Eventually the GVFS provider proceeds, and correctly populates the top-level directory, but only after a potentially long series of timeouts.

To address this we propose to find a way to, on Linux only, alter the GVFS enlistment configuration such that all paths to the `.git` directory resolve to `.gvfs/lower/.git` instead of `src/.git`.  Further, we should at the same time determine how best to set the "storage root" path of `.gvfs/lower` in a common location (i.e., somewhere in an `Enlistment` class), even though this is a platform-specific path and concept which will have no meaning on the Windows and Mac platforms.

**Comment by kivikakk** [comment]
See https://github.com/Microsoft/VFSForGit/pull/1063, https://github.com/Microsoft/VFSForGit/pull/1074.

**Comment by chrisd8088** [comment]
This is fully resolved by adjusting the path returned by [WorkingDirectoryBackingRootPath()](https://github.com/github/VFSForGit/blob/24e24df1a8370f4ed6ee09f5e91d34035e7c5058/GVFS/GVFS.Platform.Linux/LinuxPlatform.cs#L55), as @kivikakk pointed out!

This change will be incorporated into a clean PR (or two) against upstream shortly.

**Comment by chrisd8088** [comment]
Will be resolved by microsoft/VFSForGit#1125.

**Comment by kivikakk** [comment]
Closing per other tickets where the fix has been merged into microsoft#1125.
